### PR TITLE
Add support for password through ENV_VARIABLES.

### PR DESCRIPTION
**Pull request by Natim**
f? @jasonthomas r? @leplatrem

### CODE REVIEW

File: lambda.py [hunk 1]
```diff
@@ -30,15 +35,24 @@ def json2kinto(event, context):
...
     for key, value in event.items():
         if key in JSON2KINTO_ARGS:
-            args.append('--' + key)
-            args.append(value)
-
-    print("importer args", args)
-    importer_main(args)
+            args['--%s' % key] = value
+
+    flatten_args = list(reduce(lambda x, y: x + y, args.items()))
```
Comment (glasserc) [inline lambda.py]: `sum(args.items(), ())`. Why do you need this to be `list`?

**Comment by leplatrem** [comment]
I don't understand the code :( Why not just a one-liner like `value = os.getenv(key, value)` ?

File: lambda.py [hunk 2]
```diff
@@ -30,15 +35,24 @@ def json2kinto(event, context):
...
+    # Remove password from there
+    print("importer args", list(reduce(lambda x, y: x + y,
+                                       [x for x in args.items()
+                                        if x[0] not in JSON2KINTO_ENV_VARIABLES.values()])))
```
Comment (glasserc) [inline lambda.py]: Did you mean to leave this in here?

**Comment by jasonthomas** [comment]
It looks like these values will still be displayed as cleartext via the lambda console and that is not ideal. I think we need to use KMS or something else and pass a encrypted blob as the environment variable.

File: lambda.py (hunk 2 above)
Comment (Natim) [inline lambda.py]: Yes we don't want to display the secret on the command line interface.

File: lambda.py (hunk 1 above)
Comment (Natim) [inline lambda.py]: Wow `sum(args.items(), ())` seems really obscure to me. Probably a tuple would work there.

**Comment by Natim** [comment]
> it only works if no option is supposed to occur twice

It is actually the purpose of the all script.
Read the env variable and override it by the configuration if we have a duplicate.

Also the importer function is looking for a list because argparser is usually reading `sys.argv` and in this case we are providing the args to the function.

**Comment by Natim** [comment]
> It looks like these values will still be displayed as cleartext via the lambda console

No it won't because we are clearing the secret before displaying them.

**Comment by Natim** [comment]
According to https://aws.amazon.com/blogs/aws/new-for-aws-lambda-environment-variables-and-serverless-application-model/ the environment variable should be encrypted and will automatically be decrypted using KMS.

File: lambda.py (hunk 1 above)
Comment (glasserc) [inline lambda.py]: Well, [like Guido said](http://www.artima.com/weblogs/viewpost.jsp?thread=98196), there's basically only two acceptable uses of `reduce`, and they're `sum` and `product` :)

File: lambda.py (hunk 2 above)
Comment (glasserc) [inline lambda.py]: Why bother to print anything at all?

Comment (Natim) [inline lambda.py]: To make sure that the lambda runs with the proper parameters.

Comment (glasserc) [inline lambda.py]: Why does printing anything affect the way the lambda runs?

Comment (glasserc) [inline lambda.py]: Oh, I see, you mean so that you can visually confirm that the parameters are correct.

**Pull request merged**